### PR TITLE
Implement user auth

### PR DESCRIPTION
**Pull request by GregKWhite**
Adds `POST /api/v1/users/auth` endpoint. This accepts **EITHER** `email`, and `password` with `password` as the `grant_type`, **OR** `refresh_token` with `refresh_token` as the grant_type.

By default, users can have up to 20 refresh tokens. If requesting a 21st token, it will delete the user's oldest token.

Each access token is a JWT, with a lifespan of 1 hour. 

Make a request using the user's email and password to `/api/v1/users/auth` with the `password` `grant_type` to retrieve a new refresh token and access token.

Make a request using the user's refresh_token to `/api/v1/users/auth` with the `refresh_token` `grant_type` to retrieve only a new access token.

### CODE REVIEW

File: config/config.exs [hunk 1]
```diff
@@ -32,3 +32,9 @@ config :phoenix, :format_encoders,
 config :mime, :types, %{
   "application/vnd.api+json" => ["json-api"]
 }
+
+config :guardian, Guardian,
+  issuer: "MyApp",
+  ttl: { 1, :hours },
+  secret_key: %{"k" => "X5uaThfmNGtgdwaYJJjUFA", "kty" => "oct"},
```
Comment (maxcell) [inline config/config.exs:39]: I know this is a school project, but ideally would we be worried of having this committed?

Comment (GregKWhite) [reply config/config.exs:39]: Nope. We overwrite this config in a secret production file that doesn't get checked in to version control. This is only for dev and testing.

Comment (maxcell) [reply config/config.exs:39]: 👍 Sounds good to me.

**Review by maxcell — CHANGES_REQUESTED** [review]
All looks great! I tested out everything and resolved merge conflicts on my local repo, but just gotta make sure we get back in sync. Otherwise, it is good!

File: config/config.exs [hunk 2]
```diff
@@ -32,3 +32,9 @@ config :phoenix, :format_encoders,
 config :mime, :types, %{
   "application/vnd.api+json" => ["json-api"]
 }
+
+config :guardian, Guardian,
+  issuer: "MyApp",
```
Comment (GregKWhite) [inline config/config.exs]: Oh, whoops! Gotta change that to `Streamr`. I'll do that then fix merge conflicts.

**Comment by GregKWhite** [comment]
~~@maxcell Can you take one last look? I believe we're good to go now.~~

~~Fixing dogma complaints.~~

Good to go!

**Review by maxcell — APPROVED** [review]

**Comment by maxcell** [comment]
Everything looks good! I just checked through a bunch stuff. I am gonna work through a PR to do our documentation with ExDoc but if you want to go through with that, it is all good.

**Pull request merged**